Jenkinsfile

```diff
@@ -0,0 +1,542 @@
+ * see https://gitlab.com/gitlab-org/gitter/webapp/blob/master/docs/integrations.md
+ *
+ * Then you will also need to configure *global* secret text credentials containing the Gitter webhook URL,
+ * and list the ID of these credentials in the job configuration file
+ * (see https://github.com/hibernate/hibernate-jenkins-pipeline-helpers#job-configuration-file).
+ *
+ * ### Job configuration
+ *
+ * This Jenkinsfile gets its configuration from four sources:
+ * branch name, environment variables, a configuration file, and credentials.
+ * All configuration is optional for the default build (and it should stay that way),
+ * but some features require some configuration.
+ *
+ * #### Branch name
+ *
+ * See the org.hibernate.(...) imports for a link to the helpers library documentation,
+ * which explains the basics.
+ *
+ * #### Environment variables
+ *
+ * No particular environment variables is necessary.
+ *
+ * #### Job configuration file
+ *
+ * See the org.hibernate.(...) imports for a link to the helpers library documentation,
+ * which explains the basic structure of this file and how to set it up.
+ *
+ * Below is the additional structure specific to this Jenkinsfile:
+ *
+ *     deployment:
+ *       maven:
+ *         # String containing the ID of a Maven settings file registered using the config-file-provider Jenkins plugin.
+ *         # The settings must provide credentials to the servers with ID
+ *         # 'jboss-releases-repository' and 'jboss-snapshots-repository'.
+ *         settingsId: ...
+ */
+
+@Field final String MAVEN_TOOL = 'Apache Maven 3.6'
+
+// Default node pattern, to be used for resource-intensive stages.
+// Should not include the master node.
+@Field final String NODE_PATTERN_BASE = 'Slave'
+// Quick-use node pattern, to be used for very light, quick, and environment-independent stages,
+// such as sending a notification. May include the master node in particular.
+@Field final String QUICK_USE_NODE_PATTERN = 'Master||Slave'
+
+@Field AlternativeMultiMap<BuildEnvironment> environments
+@Field JobHelper helper
+
+@Field boolean enableDefaultBuild = false
+@Field boolean enableDefaultBuildIT = false
+@Field boolean performRelease = false
+@Field boolean deploySnapshot = false
+
+@Field Version releaseVersion
+@Field Version afterReleaseDevelopmentVersion
+
+this.helper = new JobHelper(this)
+
+helper.runWithNotification {
+
+stage('Configure') {
+	this.environments = AlternativeMultiMap.create([
+			jdk: [
+					// This should not include every JDK; in particular let's not care too much about EOL'd JDKs like version 9
+					// See http://www.oracle.com/technetwork/java/javase/eol-135779.html
+					new JdkBuildEnvironment(version: '8', buildJdkTool: 'OracleJDK8 Latest',
+							condition: TestCondition.BEFORE_MERGE,
+							isDefault: true),
+					new JdkBuildEnvironment(version: '11', buildJdkTool: 'OpenJDK 11 Latest',
+							condition: TestCondition.AFTER_MERGE),
```

Comment:
If they are parallelized, I think it makes sense to have both 8 and 11 running in parallel now.

Reply:
So... there is a default build, on JDK8. Other builds only run *after* that, and only if the default build succeeded. The goal is to avoid consuming too much resources if even the default build doesn't pass.

We can consider changing this behavior, but I'd rather do it as a second step. I don't know how complicated this will be, and the current behavior is the one in this Jenkinsfile anyway.